projects/cds/src/cms-components/profile-tag/profile-tag.injector.ts

```diff
@@ -0,0 +1,136 @@
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable, Renderer2 } from '@angular/core';
+import { Event as NgRouterEvent, NavigationEnd, Router } from '@angular/router';
+import {
+  AnonymousConsent,
+  AnonymousConsentsService,
+  BaseSiteService,
+  WindowRef,
+} from '@spartacus/core';
+import { fromEventPattern, merge, Observable } from 'rxjs';
+import { filter, switchMap, take, tap } from 'rxjs/operators';
+import { CdsConfig } from '../../config/cds.config';
+import {
+  ProfileTagEvent,
+  ProfileTagJsConfig,
+  ProfileTagWindowObject,
+} from './profile-tag.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProfileTagInjector {
+  static ProfileConsentTemplateId = 'PROFILE';
+  w: ProfileTagWindowObject;
+  tracking$: Observable<AnonymousConsent | NgRouterEvent>;
+  profileTagScript: HTMLScriptElement;
+  constructor(
+    private winRef: WindowRef,
+    private config: CdsConfig,
+    private baseSiteService: BaseSiteService,
+    private router: Router,
+    private anonymousConsentsService: AnonymousConsentsService,
+    @Inject(DOCUMENT) private document: Document
+  ) {
+    this.w = <ProfileTagWindowObject>(
+      (<unknown>(this.winRef ? this.winRef.nativeWindow : undefined))
+    );
+    this.w.Y_TRACKING = this.w.Y_TRACKING || {};
+    const consentChanged$ = this.consentChanged();
+    const pageLoaded$ = this.pageLoaded();
+    this.tracking$ = merge(pageLoaded$, consentChanged$);
+  }
+
+  injectScript(): Observable<AnonymousConsent | NgRouterEvent> {
+    if (!this.windowAvailable()) {
+      return;
+    }
+    return this.addTracker().pipe(
+      filter(profileTagEvent => profileTagEvent.eventName === 'Loaded'),
+      switchMap(() => {
+        return this.tracking$;
+      })
+    );
+  }
+
+  private windowAvailable() {
+    return this.w;
+  }
+
+  private pageLoaded(): Observable<NgRouterEvent> {
+    return this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd),
+      tap(() => {
+        this.w.Y_TRACKING.push({ event: 'Navigated' });
+      })
+    );
+  }
+
+  /**
+   * We are only interested in the first time the ProfileConsent is granted
+   */
+  private consentChanged(): Observable<AnonymousConsent> {
+    return this.anonymousConsentsService
+      .getConsent(ProfileTagInjector.ProfileConsentTemplateId)
+      .pipe(
+        filter(Boolean),
+        filter(profileConsent => {
+          return this.anonymousConsentsService.isConsentGiven(profileConsent);
+        }),
+        take(1),
```

Comment:
Here we are notifying profiletag that the consent has been granted. We only want to do this once, and then unsubscribe. Profiletag will then get a 'consentreference' from our system, which will represent all consents which a tenant has configured in the back end at the time the 'PROFILE' consent was granted. It will then use this for all future requests (until a client clears their cookies)
So in this case, it should be correct that we unsubscribe after the first event passes the filter. Here is the tests for this:
https://github.com/SAP/cloud-commerce-spartacus-storefront/blob/feature/ARGO-5649/projects/cds/src/cms-components/profile-tag/profile-tag.injector.spec.ts#L177